pkg/lib/authn/identity/service/service.go

```diff
@@ -65,6 +65,10 @@ type AuthenticatorService interface {
 	Delete(authenticatorInfo *authenticator.Info) error
 }
 
+type VerificationService interface {
+	RemoveOrphanedClaims(identities []*identity.Info) error
```

Comment:
Should we take secondary OOB authenticator into account? They are not associated with any identity.

Reply:
We should. However, as stated in the top post, it'll create a circular dependency between verification, identity, and authenticator, so I'm omitting it for now. There should be a TODO note in `verification/service.go`.